### DISCUSSION TITLE
Allow to control permission that can promote prompt to production level.

### DISCUSSION THREAD

**Discussion by takryo00** [langfuse/langfuse#4480] [Ideas]
### Describe the feature or potential improvement

Currently we don't have fine-grained permission to promote prompt to production.
The ability to ensure that to promote prompt to production should not occur until the approval flow is completed is required for operation in production.

**Comment by sfarthin** [comment]
Additionally, implementing a mechanism that requires users to submit relevant prompt experiments before deployment would be highly beneficial. This process would serve as an audit log, offering traceability and transparency similar to how Pull Requests manage software changes. Moreover, akin to Continuous Integration, it would provide proof of confidence that any changes are likely to improve outcomes, ensuring they are both measured and data-driven. This approach not only enhances the approval workflow, but also creates a structured record for future reference and development.

**Comment by sezf-tradify** [comment]
Yes it would be useful if:
- admins can enable/disable if approvals are required for a project
- if approvals are disabled; current behaviour is honoured
- if enabled; users who can edit a prompt can only _request_ a prompt version is promoted to production
- if enabled, a user with an approval permission is required to approve the requested change.

**Reply by marcklingen** [reply]
thanks for sharing! approvals/workflows related to prompts have been on the roadmap for a while already and demand for this increases a lot recently, will try to get to this soon

**Comment by Pringled** [comment]
Hi! This is something that would be really useful for us as well, is this on the roadmap already? Right now we have to use two separate projects (one for dev/qa, one for prod) just so we can keep permissions limited for promoting prompts to production, but ideally we would work in a single project.

**Comment by trippgrace** [comment]
it'd be great to have custom roles; we'd love to get more of our team members in Langfuse but we need more flexibility than the preset roles offer! I believe it would be a great way address this issue as well as others

**Comment by hassiebp** [comment]
We have released protected prompt labels as a first step: https://langfuse.com/changelog/2025-04-02-protected-prompt-labels
